core/network/impl/peer_manager_impl.cpp

```diff
@@ -481,4 +497,80 @@ namespace kagome::network {
     stream_engine_->add(peer_id, router_->getSupProtocol());
   }
 
+  std::vector<scale::PeerInfoSerializable>
+  PeerManagerImpl::loadLastActivePeers() {
+    auto key_res = common::Buffer::fromString(kStorageActivePeersKey);
+    if (not key_res) {
+      SL_ERROR(log_,
+               "Cannot load last active peers. Storage key cannot be "
+               "constructed. Error={}",
+               key_res.error().message());
+      return {};
+    }
+    auto &&storage_key = key_res.value();
+
+    auto get_res = storage_->get(storage_key);
+    if (not get_res) {
+      SL_ERROR(log_,
+               "List of last active peers cannot be obtained from storage. "
+               "Error={}",
+               get_res.error().message());
+      return {};
+    }
+
+    std::vector<scale::PeerInfoSerializable> last_active_peers;
+    scale::ScaleDecoderStream s{get_res.value().asVector()};
+    try {
+      s >> last_active_peers;
+    } catch (std::exception &e) {
+      SL_ERROR(log_, "Cannot decode list of active peers. Error={}", e.what());
+      return {};
+    }
+    return last_active_peers;
+  }
+
+  void PeerManagerImpl::storeActivePeers() {
+    auto key_res = common::Buffer::fromString(kStorageActivePeersKey);
+    if (not key_res) {
+      SL_ERROR(log_,
```

Comment:
And here

Reply:
Will change to assert. That is a good catch, thanks!